src/api/apiClient/apiClient.ts

```diff
@@ -0,0 +1,428 @@
+            console.error(error)
+            throw new Error(`Failed to fetch Trees: ${error.message}`);
+        }
+    }
+
+    async createTree(data: Tree): Promise<Tree> {
+        try {
+            const response = await axios.post<Tree>(`/trees/`, data);
+            return response.data;
+        } catch (error) {
+            console.error(error)
+            throw new Error('Failed to create Tree');
+        }
+    }
+
+    async updateTree(data: Tree): Promise<Tree> {
+        try {
+            const response = await axios.put<Tree>(`/trees/${data._id}`, data);
+            return response.data;
+        } catch (error) {
+            console.error(error)
+            throw new Error('Failed to update Tree');
+        }
+    }
+
+    async deleteTree(data: Tree): Promise<string> {
+        try {
+            await axios.delete<any>(`/trees/${data._id}`);
+            return data._id;
+        } catch (error) {
+            console.error(error)
+            throw new Error('Failed to delete Tree');
+        }
+    }
+
+    async createTreeBulk(data: Blob): Promise<void> {
+        try {
+            const formData = new FormData();
+            formData.append('csvFile', data, 'trees_data.csv');
+            await axios.post<any>(`/trees/bulk`, );
+        } catch (error) {
+            console.error(error)
+            throw new Error('Failed to create Trees in bulk');
+        }
+    }
+
+    /*
+        Model- UserTree: CRUD Operations/Apis for user_tree_regs
+    */
+
+    async getUserTrees(): Promise<UserTree[]> {
+        const url = `/profile/`;
+        try {
+            const response = await this.api.get<UserTree[]>(url);
+            return response.data;
+        } catch (error: any) {
+            console.error(error)
+            throw new Error(`Failed to fetch user tree profile: ${error.message}`);
+        }
+    }
+
+    async createUserTree(data: UserTree): Promise<UserTree> {
+        try {
+            const response = await axios.post<UserTree>(`/profile/`, data);
+            return response.data;
+        } catch (error) {
+            console.error(error)
+            throw new Error('Failed to create user tree profile');
+        }
+    }
+
+    async updateUserTree(data: UserTree): Promise<UserTree> {
+        try {
+            const response = await axios.put<UserTree>(`/profile/${data._id}`, data);
+            return response.data;
+        } catch (error) {
+            console.error(error)
+            throw new Error('Failed to update user tree profile');
+        }
+    }
+
+    async deleteUserTree(data: UserTree): Promise<string> {
+        try {
+            await axios.delete<any>(`/profile/${data._id}`);
+            return data._id;
+        } catch (error) {
+            console.error(error)
+            throw new Error('Failed to delete user tree profile');
+        }
+    }
+    
+}
+
+// new function to fetch data form localStorage
+export const fetchDataFromLocal = (key: string) => {
```

Comment:
Are we using localStorage for all api responses? 
Seems like a premature optimisation to me, unless there are some better reasons I'm missing. 
Just want to make sure we don't make the app harder to debug.